### ISSUE TITLE
zypper ignores --no-recommends

### ISSUE THREAD

**Issue by bmwiedemann** [openSUSE/zypper#459]
On Leap 15.4 and 15.3, I found that `zypper up --no-recommends`
would still propose to install recommended packages (`wireless-regdb` among those)
`installRecommends = no` in /etc/zypp/zypper.conf is ignored as well, but `solver.onlyRequires = true` in `zypp.conf` still works. `zypper addlock` also works as expected.

```
rpm -q zypper libzypp 
zypper-1.14.55-150400.3.6.1.x86_64
libzypp-17.31.0-150400.3.6.1.x86_64

#15.3:
zypper-1.14.53-150200.33.1.x86_64
libzypp-17.30.2-150200.39.1.x86_64
```

With a quick bisect I found that zypper-1.14.52 still works fine, but zypper-1.14.53 and later has this problem.

**Comment by bmwiedemann** [comment]
`git bisect` pointed at commit 1d7698b7581e8c9b60f8c49e181e81f51052a6d5

> There should be no more need to explicitely establish the PPP status

**Comment by mlandres** [comment]
Fixed in libzypp-17.31.2
